modules/api/pkg/ee/resource-quota/handler.go

```diff
@@ -172,35 +214,373 @@ func GetResourceQuota(ctx context.Context, request interface{}, provider provide
+		if err = getVMCloudDirectorResourceDetails(req, nc); err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("provider set in request not supported: %v", req.Body)
+	}
+
+	// These should not happen, unless we have a bug in the setting resource values code above
+	if nc.CPUCores == nil {
+		return nil, fmt.Errorf("cpu not set in the request: %v", req.Body)
+	}
+	if nc.Memory == nil {
+		return nil, fmt.Errorf("memory not set in the request: %v", req.Body)
+	}
+	if nc.Storage == nil {
+		return nil, fmt.Errorf("storage not set in the request: %v", req.Body)
+	}
+
+	// Multiply by replicas count
+	var cpu, mem, sto resource.Quantity
+	for i := 0; i < req.Body.Replicas; i++ {
+		cpu.Add(*nc.CPUCores)
+		mem.Add(*nc.Memory)
+		sto.Add(*nc.Storage)
+	}
+
+	rd := kubermaticv1.NewResourceDetails(cpu, mem, sto)
+
+	return rd, nil
+}
+
+func getAlibabaResourceDetails(req calculateProjectResourceQuotaUpdate, nc *kubermaticprovider.NodeCapacity) error {
+	nc.WithCPUCount(req.Body.AlibabaInstanceType.CPUCoreCount)
+
+	if err := nc.WithMemory(int(req.Body.AlibabaInstanceType.MemorySize), "G"); err != nil {
+		return err
+	}
+	if err := nc.WithStorage(req.Body.DiskSizeGB, "G"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func getAnexiaResourceDetails(req calculateProjectResourceQuotaUpdate, nc *kubermaticprovider.NodeCapacity) error {
+	nc.WithCPUCount(req.Body.AnexiaNodeSpec.CPUs)
+
+	if err := nc.WithMemory(int(req.Body.AnexiaNodeSpec.Memory), "M"); err != nil {
+		return err
+	}
+
+	var diskSize int64
+	for _, disk := range req.Body.AnexiaNodeSpec.Disks {
+		diskSize += disk.Size
+	}
+	if err := nc.WithStorage(int(diskSize), "G"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func getAWSResourceDetails(req calculateProjectResourceQuotaUpdate, nc *kubermaticprovider.NodeCapacity) error {
+	nc.WithCPUCount(req.Body.AWSSize.VCPUs)
+
+	if err := nc.WithMemory(int(req.Body.AWSSize.Memory), "G"); err != nil {
+		return err
+	}
+	if err := nc.WithStorage(req.Body.DiskSizeGB, "G"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func getAzureResourceDetails(req calculateProjectResourceQuotaUpdate, nc *kubermaticprovider.NodeCapacity) error {
+	nc.WithCPUCount(int(req.Body.AzureSize.NumberOfCores))
+
+	if err := nc.WithMemory(int(req.Body.AzureSize.MemoryInMB), "M"); err != nil {
+		return err
+	}
+
+	if err := nc.WithStorage(int(req.Body.AzureSize.ResourceDiskSizeInMB+req.Body.AzureSize.OsDiskSizeInMB), "M"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func getDOResourceDetails(req calculateProjectResourceQuotaUpdate, nc *kubermaticprovider.NodeCapacity) error {
+	nc.WithCPUCount(req.Body.DOSize.VCPUs)
+
+	if err := nc.WithMemory(req.Body.DOSize.Memory, "M"); err != nil {
+		return err
+	}
+	if err := nc.WithStorage(req.Body.DOSize.Disk, "G"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func getEquinixResourceDetails(req calculateProjectResourceQuotaUpdate, nc *kubermaticprovider.NodeCapacity) error {
+	var cpuCount int
```

Comment:
nit: I know it's golang, but I would still prefer to init this variable explicitly with 0.
```suggestion
	cpuCount := 0
```